src/views/List.jsx

```diff
@@ -1,13 +1,48 @@
+import { useEffect, useState } from 'react';
 import { ListItem } from '../components';
 
 export function List({ data }) {
+	const [searchInput, setSearchInput] = useState('');
+	const [filteredData, setFilteredData] = useState([]);
+
+	console.log('filtered Data', filteredData);
+
+	useEffect(() => {
+		setFilteredData(data);
+	}, [data]);
+
+	const handleSearchInput = (e) => {
+		const text = e.target.value.toLowerCase();
+		setSearchInput(text);
+		setFilteredData(
+			data.filter((item) => item.name.toLowerCase().includes(text)),
```

Comment:
I think it would make more sense to move the `toLowerCase()` from when you declare `text` down the copy of `text` inside the `includes` so that if the user wants to type in uppercase it will show up in uppercase instead of always being lowercase, like so:

`data.filter((item) => item.name.toLowerCase().includes(text.toLowerCase())),`

Otherwise it works great!